Mp3 files are recognized as a certificate or (nfc) tag

Steps to reproduce: just take any file, and rename it to filename.mp3 then try to open it. It will show you a few apps to open it. (like the certificate manager or the Tag manager) open the file with a app, and you get the error. 





## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.2
* __OS:__ Linux Android 10 - 29
* __Device:__ a51
* __Model:__ SM-A515F
* __Product:__ a51nseea
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: Permission Denial: starting Intent { act=android.intent.action.VIEW dat=content://com.amaze.filemanager/storage_root/storage/emulated/0/Download/stdout.mp3 typ=*/* flg=0x1 cmp=com.android.apps.tag/.TagViewer } from ProcessRecord{74f7f71 3139:com.amaze.filemanager/u0a510} (pid=3139, uid=10510) requires android.permission.NFC
	at android.os.Parcel.createException(Parcel.java:2088)
	at android.os.Parcel.readException(Parcel.java:2056)
	at android.os.Parcel.readException(Parcel.java:2004)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:4454)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1716)
	at android.app.Activity.startActivityForResult(Activity.java:5258)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:676)
	at android.app.Activity.startActivityForResult(Activity.java:5203)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:663)
	at android.app.Activity.startActivity(Activity.java:5587)
	at android.app.Activity.startActivity(Activity.java:5555)
	at com.amaze.filemanager.ui.dialogs.OpenFileDialogFragment$loadViews$$inlined$let$lambda$1.onClick(OpenFileDialogFragment.kt:344)
	at android.view.View.performClick(View.java:7862)
	at android.widget.TextView.performClick(TextView.java:15004)
	at android.view.View.performClickInternal(View.java:7831)
	at android.view.View.access$3600(View.java:879)
	at android.view.View$PerformClick.run(View.java:29359)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8167)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.wm.ActivityStackSupervisor.checkStartAnyActivityPermission(ActivityStackSupervisor.java:1447)
	at com.android.server.wm.ActivityStarter.startActivity(ActivityStarter.java:980)
	at com.android.server.wm.ActivityStarter.startActivity(ActivityStarter.java:732)
	at com.android.server.wm.ActivityStarter.startActivityMayWait(ActivityStarter.java:2074)
	at com.android.server.wm.ActivityStarter.execute(ActivityStarter.java:646)


```
</details>
<hr>

--- a/app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java
+++ b/app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java
@@ -600,8 +600,8 @@ public class FileUtils {
     boolean useNewStack =
         sharedPrefs.getBoolean(PreferencesConstants.PREFERENCE_TEXTEDITOR_NEWSTACK, false);
     boolean defaultHandler = isSelfDefault(f, m);
-    SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(m);
     final Toast[] studioCount = {null};
+    final int studio_count = PreferenceManager.getDefaultSharedPreferences(m).getInt("studio", 0);
 
     if (f.getName().toLowerCase().endsWith(".apk")) {
       GeneralDialogCreation.showPackageDialog(f, m);
@@ -611,35 +611,26 @@ public class FileUtils {
       Intent intent = new Intent(m, DatabaseViewerActivity.class);
       intent.putExtra("path", f.getPath());
       m.startActivity(intent);
-    } else if (Icons.getTypeOfFile(f.getPath(), f.isDirectory()) == Icons.AUDIO) {
-      final int studio_count = sharedPreferences.getInt("studio", 0);
-      Uri uri = Uri.fromFile(f);
-      final Intent intent = new Intent();
-      intent.setAction(Intent.ACTION_VIEW);
-      intent.setDataAndType(uri, "audio/*");
-
+    } else if (Icons.getTypeOfFile(f.getPath(), f.isDirectory()) == Icons.AUDIO
+        && studio_count != 0) {
       // Behold! It's the  legendary easter egg!
-      if (studio_count != 0) {
-        new CountDownTimer(studio_count, 1000) {
-          @Override
-          public void onTick(long millisUntilFinished) {
-            int sec = (int) millisUntilFinished / 1000;
-            if (studioCount[0] != null) studioCount[0].cancel();
-            studioCount[0] = Toast.makeText(m, sec + "", Toast.LENGTH_LONG);
-            studioCount[0].show();
-          }
+      new CountDownTimer(studio_count, 1000) {
+        @Override
+        public void onTick(long millisUntilFinished) {
+          int sec = (int) millisUntilFinished / 1000;
+          if (studioCount[0] != null) studioCount[0].cancel();
+          studioCount[0] = Toast.makeText(m, sec + "", Toast.LENGTH_LONG);
+          studioCount[0].show();
+        }
 
-          @Override
-          public void onFinish() {
-            if (studioCount[0] != null) studioCount[0].cancel();
-            studioCount[0] = Toast.makeText(m, m.getString(R.string.opening), Toast.LENGTH_LONG);
-            studioCount[0].show();
-            openFileDialogFragmentFor(f, m, "audio/*");
-          }
-        }.start();
-      } else {
-        openFileDialogFragmentFor(f, m);
-      }
+        @Override
+        public void onFinish() {
+          if (studioCount[0] != null) studioCount[0].cancel();
+          studioCount[0] = Toast.makeText(m, m.getString(R.string.opening), Toast.LENGTH_LONG);
+          studioCount[0].show();
+          openFileDialogFragmentFor(f, m, "audio/*");
+        }
+      }.start();
     } else {
       try {
         openFileDialogFragmentFor(f, m);

--- a/app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java
+++ b/app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java
@@ -634,31 +634,47 @@ public class FileUtils {
             if (studioCount[0] != null) studioCount[0].cancel();
             studioCount[0] = Toast.makeText(m, m.getString(R.string.opening), Toast.LENGTH_LONG);
             studioCount[0].show();
-            OpenFileDialogFragment.Companion.openFileOrShow(
-                FileProvider.getUriForFile(m, m.getPackageName(), f),
-                "audio/*",
-                useNewStack,
-                m,
-                false);
+            openFileDialogFragmentFor(f, m, "audio/*");
           }
         }.start();
       } else {
-        OpenFileDialogFragment.Companion.openFileOrShow(
-            FileProvider.getUriForFile(m, m.getPackageName(), f), "*/*", useNewStack, m, false);
+        openFileDialogFragmentFor(f, m);
       }
     } else {
       try {
-        OpenFileDialogFragment.Companion.openFileOrShow(
-            FileProvider.getUriForFile(m, m.getPackageName(), f),
-            MimeTypes.getMimeType(f.getAbsolutePath(), false),
-            useNewStack,
-            m,
-            false);
+        openFileDialogFragmentFor(f, m);
       } catch (Exception e) {
         Toast.makeText(m, m.getString(R.string.no_app_found), Toast.LENGTH_LONG).show();
         openWith(f, m, useNewStack);
       }
     }
+  }
+
+  private static void openFileDialogFragmentFor(
+      @NonNull File file, @NonNull MainActivity mainActivity) {
+    openFileDialogFragmentFor(
+        file, mainActivity, MimeTypes.getMimeType(file.getAbsolutePath(), false));
+  }
+
+  private static void openFileDialogFragmentFor(
+      @NonNull File file, @NonNull MainActivity mainActivity, @NonNull String mimeType) {
+    OpenFileDialogFragment.Companion.openFileOrShow(
+        FileProvider.getUriForFile(mainActivity, mainActivity.getPackageName(), file),
+        mimeType,
+        false,
+        mainActivity,
+        false);
+  }
+
+  private static void openFileDialogFragmentFor(
+      @NonNull DocumentFile file, @NonNull MainActivity mainActivity) {
+    openFileDialogFragmentFor(
+        file.getUri(), mainActivity, MimeTypes.getMimeType(file.getUri().toString(), false));
+  }
+
+  private static void openFileDialogFragmentFor(
+      @NonNull Uri uri, @NonNull MainActivity mainActivity, @NonNull String mimeType) {
+    OpenFileDialogFragment.Companion.openFileOrShow(uri, mimeType, false, mainActivity, false);
   }
 
   private static boolean isSelfDefault(File f, Context c) {
@@ -679,8 +695,7 @@ public class FileUtils {
     boolean useNewStack =
         sharedPrefs.getBoolean(PreferencesConstants.PREFERENCE_TEXTEDITOR_NEWSTACK, false);
     try {
-      OpenFileDialogFragment.Companion.openFileOrShow(
-          f.getUri(), MimeTypes.getMimeType(f.getUri().toString(), false), useNewStack, m, false);
+      openFileDialogFragmentFor(f, m);
     } catch (Exception e) {
       Toast.makeText(m, m.getString(R.string.no_app_found), Toast.LENGTH_LONG).show();
       openWith(f, m, useNewStack);

